access_group_cache/main.py

```diff
@@ -14,46 +15,70 @@
 app = Flask(__name__)
 
 
-async def _groups_lookup(access_token: str, group_name: str) -> Optional[str]:
+async def _get_group_parent(access_token: str, group_name: str) -> Optional[str]:
+    """
+    Get the group parent (group ID)
+    """
+    if group_name.endswith('.iam.gserviceaccount.com'):
+        # we know it's not a group, because it's a service account
+        return None
+
     async with aiohttp.ClientSession() as session:
         # https://cloud.google.com/identity/docs/reference/rest/v1/groups/lookup
         async with session.get(
             f'https://cloudidentity.googleapis.com/v1/groups:lookup?'
             f'groupKey.id={urllib.parse.quote(group_name)}',
             headers={'Authorization': f'Bearer {access_token}'},
         ) as resp:
-            if resp.status != 200:
+            if resp.status == 403:
```

Comment:
Should this ever happen if we already have the filter at the top? Should this instead raise an assertion that contains more info (is there additional status text in `resp` for the 403 case?)?

Reply:
The filter at the top only stops service accounts, users can still be members of the group and they're harder to sanity check. The new 'group' check in the group_membership function should now (I think) stop all these from coming through this channel, but I don't know how to get 100% confidence that it will.

There's no extra information in the response, just `https://cloudidentity.googleapis.com/v1/groups:lookup?groupKey.id=access-group-cache@analysis-runner.iam.gserviceaccount.com`